src/routes/transactions/tests/routes.rs

```diff
@@ -0,0 +1,429 @@
+extern crate dotenv;
+
+use crate::common::models::page::Page;
+use crate::config::{
+    chain_info_request_timeout, contract_info_request_timeout, safe_info_request_timeout,
+    token_info_request_timeout, transaction_request_timeout,
+};
+use crate::providers::info::TokenInfo;
+use crate::routes::transactions::models::details::TransactionDetails;
+use crate::routes::transactions::tests::{MULTISIG_TX_DETAILS, POST_CONFIRMATION_RESULT};
+use crate::tests::main::setup_rocket;
+use crate::utils::errors::{ApiError, ErrorDetails};
+use crate::utils::http_client::{MockHttpClient, Request, Response};
+use core::time::Duration;
+use mockall::predicate::eq;
+use rocket::http::{ContentType, Header, Status};
+use rocket::local::asynchronous::Client;
+use serde_json::json;
+
+#[rocket::async_test]
+async fn post_confirmation_success() {
+    let mock_http_client = {
+        let mut mock_http_client = MockHttpClient::new();
+
+        let mut chain_request = Request::new(config_uri!("/v1/chains/{}/", 4));
+        chain_request.timeout(Duration::from_millis(chain_info_request_timeout()));
+        mock_http_client
+            .expect_get()
+            .times(1)
+            .with(eq(chain_request))
+            .returning(move |_| {
+                Ok(Response {
+                    status_code: 200,
+                    body: String::from(crate::tests::json::CHAIN_INFO_RINKEBY),
+                })
+            });
+
+        // CONFIRMATION REQUEST
+        let mut backend_request = Request::new(
+                "https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/multisig-transactions/0x2e4af4b451a493470f38625c5f78f710f02303eb32780896cb55357c00d48faa/confirmations/"
+                .to_string(),
+        );
+        backend_request.body(Some(json!({"signature": "bd42f5c205b544cc6397c8c2e592ca4ade02b8681673cc8c555ff1777b002ee959c3cca243a77a2de1bbe1b61413342ac7d6416a31ec0ff31bb1029e921202ee1c"}).to_string()));
```

Comment:
We should look into creating factories for most of these objects. Right now we are mostly relying on data from a json file and it can be more tricky to see why we need values like `0x2e4af4b451a493470f38625c5f78f710f02303eb32780896cb55357c00d48faa` and `bd42f5c205b544cc6397c8c2e592ca4ade02b8681673cc8c555ff1777b002ee959c3cca243a77a2de1bbe1b61413342ac7d6416a31ec0ff31bb1029e921202ee1c`.

With a Factory, we'd generate random (valid) data for all the fields except for the ones which actually matter for the test.

Reply:
I would love to explore this further. I think it is going to be easier to define these reqs as we write more tests. But it is defintely something we need to tackle early on so we don't litter the code too much.